tests/test_D2_model_constraints.py

```diff
@@ -225,46 +237,56 @@ def test_add_constraints_maximum_emissions_None(self):
             {
                 CONSTRAINTS: {
                     MAXIMUM_EMISSIONS: {VALUE: None},
-                    MINIMAL_RENEWABLE_FACTOR: {
-                        VALUE: self.dict_values[CONSTRAINTS][MINIMAL_RENEWABLE_FACTOR][
-                            VALUE
-                        ]
-                    },
+                    MINIMAL_RENEWABLE_FACTOR: {VALUE: 0},
                 }
             }
         )
         model = D2.add_constraints(
-            local_energy_system=self.model,
-            dict_values=self.dict_values,
+            local_energy_system=solph.Model(self.model),
+            dict_values=dict_values,
             dict_model=self.dict_model,
         )
         assert (
-            model.integral_limit_emission_factor.NoConstraint[0]
-            == self.exp_emission_limit
+            hasattr(model, "integral_limit_emission_factor") == False
         ), f"When maximum_emission is None, no emission constraint should be added to the ESM."
 
-    """
+    # This test is to check if the constraint minimal renewable share value provided by the user is being applied or not
     def test_add_constraints_minimal_renewable_share(self):
-    # todo to be added
-    pass
-    """
+        dict_values = self.dict_values.copy()
 
-    """
-    def test_add_constraints_minimal_renewable_share_None(self):
+        model = D2.add_constraints(
+            local_energy_system=solph.Model(self.model),
+            dict_values=dict_values,
+            dict_model=self.dict_model,
+        )
+
+        assert (
+            hasattr(model, "constraint_minimal_renewable_share") == True
+        ), f"The minimal renewable share has not been added, something has failed."
+
+    # Test to make sure that the minimal renewable share constraint is not being applied if the user does not provide any value for it
+    def test_add_constraints_minimal_renewable_share_is_0(self):
         dict_values = self.dict_values.copy()
         dict_values.update(
             {
                 CONSTRAINTS: {
                     MAXIMUM_EMISSIONS: {
                         VALUE: self.dict_values[CONSTRAINTS][MAXIMUM_EMISSIONS][VALUE]
                     },
```

Comment:
I guess this part above, setting the maximum_emissions could be deleted (no interest in that in this test). Alternatively you could set the value to `None` to not add this constraint in this test.

Reply:
Done :)